### PR TITLE
feat: ARM-arch is now supported, avoid always recomputing values

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: core/src/main/java/io/javaoperatorsdk/jenvtest/binary/OSInfo.java [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
-  public String getOSName() {
-    String os = System.getProperty("os.name").toLowerCase();
+  public OSInfo() {
+    final var osArch = System.getProperty("os.arch").toLowerCase();
+    switch (osArch) {
+      case "ppc64":
+        arch = "ppc64le";
+        break;
+      case "aarch64":
+        arch = "arm64";
```
Comment (csviri) [inline core/src/main/java/io/javaoperatorsdk/jenvtest/binary/OSInfo.java:14]: there are no binaries for ARM, but on macs this should work with amd64 binaries.

**Review by csviri — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/javaoperatorsdk/jenvtest/binary/OSInfo.java (hunk 1 above)
Comment (metacosm) [reply core/src/main/java/io/javaoperatorsdk/jenvtest/binary/OSInfo.java:14]: There are binaries for ARM and, no, `amd64` won't work on more recent Macs. I've checked on both.

Comment (csviri) [reply core/src/main/java/io/javaoperatorsdk/jenvtest/binary/OSInfo.java:14]: ahh, sorry your are right, I completely missed that

**Review by csviri — APPROVED** [review]

**Comment by csviri** [comment]
would be nice to have an automated test run for this, just probably not possible now with github actions, or IDK

**Pull request merged**